core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -124,10 +124,16 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
     (blockId, getFile(blockId))
   }
 
+  /**
+   * Create local directories for storing block data. These directories are
+   * located inside configured local directories and won't
+   * be deleted on JVM exit when using the external shuffle service.
```

Comment:
I'm pretty confused. When external shuffle service is enabled, the executors already do not clean up the shuffle files when the JVM exits.